ambry-replication/src/test/java/com.github.ambry.replication/ReplicationTest.java

```diff
@@ -309,6 +321,182 @@ public void replicationPauseTest() throws Exception {
+        B0'     B0
+        B1'     B1
+                B2
+        B0 is B0' for local,
+        B1 is B1' for local,
+        B2 is null for local,
+        so it already has B0/B0'
+        B1 is transferred to B1'
+        and B2 is invalid for L
+        so it does not count as missing
+        Missing Keys: 1
+
+     */
+    Map<PartitionId, List<BlobId>> partitionIdToDeleteBlobId = new HashMap<>();
+    Map<StoreKey, StoreKey> conversionMap = new HashMap<>();
+    Map<PartitionId, BlobId> expectedPartitionIdToDeleteBlobId = new HashMap<>();
+    for (int i = 0; i < partitionIds.size(); i++) {
+      PartitionId partitionId = partitionIds.get(i);
+      List<BlobId> deleteBlobIds = new ArrayList<>();
+      partitionIdToDeleteBlobId.put(partitionId, deleteBlobIds);
+      BlobId b0 = generateRandomBlobId(partitionId);
+      deleteBlobIds.add(b0);
+      BlobId b0p = generateRandomBlobId(partitionId);
+      expectedPartitionIdToDeleteBlobId.put(partitionId, b0p);
+      BlobId b1 = generateRandomBlobId(partitionId);
+      BlobId b1p = generateRandomBlobId(partitionId);
+      BlobId b2 = generateRandomBlobId(partitionId);
+      deleteBlobIds.add(b2);
+      conversionMap.put(b0, b0p);
+      conversionMap.put(b1, b1p);
+      conversionMap.put(b2, null);
+      //Convert current conversion map so that BlobIdTransformer can
+      //create b1p in expectedLocalHost
+      storeKeyConverter.setConversionMap(conversionMap);
+      storeKeyConverter.convert(conversionMap.keySet());
+      addPutMessagesToReplicasOfPartition(Arrays.asList(b0p), Arrays.asList(localHost));
+      addPutMessagesToReplicasOfPartition(Arrays.asList(b0, b1, b2), Arrays.asList(remoteHost));
+      addPutMessagesToReplicasOfPartition(Arrays.asList(b0p, b1), Arrays.asList(null, transformer),
+          Arrays.asList(expectedLocalHost));
+
+      //Check that expected local host contains the correct blob ids
+      Set<BlobId> expectedLocalHostBlobIds = new HashSet<>();
+      expectedLocalHostBlobIds.add(b0p);
+      expectedLocalHostBlobIds.add(b1p);
+      assertEquals("Should be two ids in expectedLocalHostBlobIds", 2, expectedLocalHostBlobIds.size());
+      for (MessageInfo messageInfo : expectedLocalHost.infosByPartition.get(partitionId)) {
+        assertTrue("Remove should never fail", expectedLocalHostBlobIds.remove(messageInfo.getStoreKey()));
+      }
+      assertTrue("expectedLocalHostBlobIds should now be empty", expectedLocalHostBlobIds.isEmpty());
+    }
+    storeKeyConverter.setConversionMap(conversionMap);
+
+    expectedIndex = assertMissingKeysAndFixMissingStoreKeys(expectedIndex, 2, batchSize, 1, replicaThread, remoteHost,
+        replicasToReplicate);
+
+    //Check that there are no missing buffers between expectedLocalHost and LocalHost
+    Map<PartitionId, List<ByteBuffer>> missingBuffers =
+        expectedLocalHost.getMissingBuffers(localHost.buffersByPartition);
+    for (List<ByteBuffer> list : missingBuffers.values()) {
```

Comment:
Looking at `getMissingBuffers()`, it looks to me that you should be checking for `missingBuffers.isEmpty()` instead - here and everywhere. It does not seem to set empty lists for partitions.

Reply:
you're right, changed